### DISCUSSION TITLE
Nuxt 3 + Typescript initial setup

### DISCUSSION THREAD

**Discussion by ghost** [nuxt/nuxt#18785] [Questions] [answered]
Hello, I just installed Nuxt 3 and got multiple questions.
Do I have to change some settings to fully use TS? As for now app can use JS and TS too if lang='ts', but here are some problems:

![1](https://user-images.githubusercontent.com/103945399/216839075-bc232708-c81c-41de-8e46-db15805071ab.jpg)
For some reason function is highlighted with red line, with this error: 'property someFn' doesnt exist on type {}'

![Screenshot 2023-02-05 at 22 54 30](https://user-images.githubusercontent.com/103945399/216839343-78db13df-f5aa-4395-95e4-77035e122cb1.jpg)

Also here, tsconfig.json is erroring opening bracket with error: 'JSON schema for the TypeScript compiler's configuration file'.

So do I need to make some changes in initial project for TS?

**Comment by AloisSeckar** [comment, marked as answer]
There shouldn't be any extra in-application setup. Nuxt app runs even with no config file at all.

I briefly saw the same error as you have after I deleted my `node_modules` folder and before `npm install` finished running. So maybe try `npm install` first to see, if you aren't missing some dependencies yet? 

Also do you have [Volar plugin for VSCode](https://marketplace.visualstudio.com/items?itemName=Vue.volar) installed and enanbled for your workspace? From syntax highlighting I would say so, but to make sure...


Last but not least there is an error in your function call - you decared `val` to be `string` but you're trying to pass a `number`. TS won't be happy with that once it grabs the function definition correctly ;)

**Reply by ghost** [reply]
Yes, I've installed volar, and even with string error still persists. npm install and refreshing my IDE solved the problem, thanks!
